app/src/main/java/pl/futurecollars/invoicing/service/InvoiceService.java

```diff
@@ -0,0 +1,39 @@
+package pl.futurecollars.invoicing.service;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+
+@Service
+public class InvoiceService {
+
+    private final Database database;
+
+    public InvoiceService(@Qualifier("fileDatabase") Database database) {
+        this.database = database;
+    }
+
+    public int save(Invoice invoice) {
+        return database.save(invoice);
+    }
+
+    public Optional<Invoice> getById(int id) {
+        return database.getById(id);
+    }
+
+    public List<Invoice> getAlL() {
+        return database.getAlL();
+    }
+
+    public Optional<Invoice> update(int id, Invoice updatedInvoice) {
+        database.update(id, updatedInvoice);
+        return null;
```

Comment:
Jeśli chodzi o controllera, to update nie działa bo metoda zwraca nulla...:P

Reply:
tak.... już zwraca prawidłowy Content i działa :)